### PR TITLE
Capture failure-logs in build diagnostics

### PR DESCRIPTION
**Pull request by pshipton**

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins compile zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

**Comment by pshipton** [comment]
jenkins compile zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

**Comment by pshipton** [comment]
This is working, the diagnostics file for the failed build contains the failure-logs directory.
https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDKnext_s390x_linux_Personal/154/Build_JDKnext_s390x_linux_Personal-154-20210406-111741-diagnostics.tar.gz

File: buildenv/jenkins/common/build.groovy [hunk 1]
```diff
@@ -471,7 +471,7 @@ def archive_diagnostics() {
         // Note: to preserve the files ACLs set _OS390_USTAR=Y env variable (see variable files)
         sh "find . -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' | sed 's#^./##' | pax -wzf ${diagnosticsFilename}"
     } else {
-        sh "find . -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' -o -name ddr_info -o -name j9ddr.dat -o -name superset.dat -o -name '*.annt.h' -o -name '*.stub.h' -o -name '*.dSYM' | sed 's#^./##' | tar -zcvf ${diagnosticsFilename} -T -"
+        sh "find . -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' -o -name ddr_info -o -name j9ddr.dat -o -name superset.dat -o -name '*.annt.h' -o -name '*.stub.h' -o -name '*.dSYM' -o -wholename '*/make-support/failure-logs/*' | sed 's#^./##' | tar -zcvf ${diagnosticsFilename} -T -"
```
Comment (keithc-ca) [inline buildenv/jenkins/common/build.groovy:474]: According to `man find`, 'This alternative is less portable than `-path`', which is 'part of the POSIX 2008 standard'.

**Comment by keithc-ca** [comment]
BTW, the errors captured in that failed build are fixed by https://github.com/eclipse/openj9/pull/12360.

**Comment by keithc-ca** [comment]
Also, this isn't specific to cmake.

**Comment by pshipton** [comment]
> Also, this isn't specific to cmake.

Not sure if you are saying a change to the PR is needed.

**Comment by pshipton** [comment]
jenkins compile zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

File: buildenv/jenkins/common/build.groovy (hunk 1 above)
Comment (pshipton) [reply buildenv/jenkins/common/build.groovy]: Updated, testing it now.

**Comment by pshipton** [comment]
> Also, this isn't specific to cmake.

Got it, fixed the commit title.

**Comment by pshipton** [comment]
The running PR test is https://ci.eclipse.org/openj9/view/Pull%20Requests/job/PullRequest-OpenJ9/5752/

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
The change to use `-path` is working.
https://ci.eclipse.org/openj9/job/Build_JDKnext_s390x_linux_Personal/156

https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDKnext_s390x_linux_Personal/156/Build_JDKnext_s390x_linux_Personal-156-20210406-122924-diagnostics.tar.gz

**Pull request merged**